contracts/modules/cover/Cover.sol

```diff
@@ -316,28 +357,79 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
     }
   }
 
-  function requestAllocation(
+  function requestAllocationsForNewCover(
     AllocationRequest memory allocationRequest,
     PoolAllocationRequest[] memory poolAllocationRequests,
-    uint nxmPriceInCoverAsset,
-    uint segmentId
+    uint nxmPriceInCoverAsset
   ) internal returns (
     uint totalCoverAmountInCoverAsset,
     uint totalAmountDueInNXM
   ) {
 
-    RequestAllocationVariables memory vars = RequestAllocationVariables(0, 0, 0, 0);
     uint totalCoverAmountInNXM;
 
-    vars.previousPoolAllocationsLength = segmentId > 0
-      ? coverSegmentAllocations[allocationRequest.coverId][segmentId - 1].length
-      : 0;
+    for (uint i = 0; i < poolAllocationRequests.length; i++) {
+
+      // converting asset amount to nxm and rounding up to the nearest NXM_PER_ALLOCATION_UNIT
+      uint coverAmountInNXM = getNXMForAssetAmount(
+        poolAllocationRequests[i].coverAmountInAsset,
+        nxmPriceInCoverAsset
+      );
+
+    (uint premiumInNXM, uint allocationId) = stakingProducts().stakingPool(poolAllocationRequests[i].poolId).requestAllocation(
+        coverAmountInNXM,
+        0, // previousAllocationAmountInNXMRepriced
+        allocationRequest
+      );
+
+      // TODO: consider what happens if user creates lots of segments with small amounts
+      coverSegmentAllocations[allocationRequest.coverId][0].push(
+        PoolAllocation(
+          poolAllocationRequests[i].poolId,
+          coverAmountInNXM.toUint96(),
+          premiumInNXM.toUint96(),
+          allocationId.toUint24()
+        )
+      );
+
+      totalAmountDueInNXM += premiumInNXM;
+      totalCoverAmountInNXM += coverAmountInNXM;
+    }
+
+    totalCoverAmountInCoverAsset = totalCoverAmountInNXM * nxmPriceInCoverAsset / ONE_NXM;
+
+    return (totalCoverAmountInCoverAsset, totalAmountDueInNXM);
+  }
+
+  function requestEditAllocations(
+    AllocationRequest memory allocationRequest,
+    PoolAllocationRequest[] memory poolAllocationRequests,
+    AllocationParams memory params
+  ) internal returns (
+    uint totalCoverAmountInCoverAsset,
+    uint totalAmountDueInNXM
+  ) {
+
+    RequestAllocationVariables memory vars; // = RequestAllocationVariables(0, 0, 0, 0);
+
+    vars.previousPoolAllocationsLength = coverSegmentAllocations[allocationRequest.coverId][params.segmentId - 1].length;
+
+    vars.previousPoolAllocations = new PoolAllocation[](vars.previousPoolAllocationsLength);
+
+    for (uint i = 0; i < vars.previousPoolAllocationsLength; i++) {
+      vars.previousPoolAllocations[i] = coverSegmentAllocations[allocationRequest.coverId][params.segmentId - 1][i];
+
+      vars.previousTotalCoverAmountInNXM += vars.previousPoolAllocations[i].coverAmountInNXM;
+    }
+
+    uint oldSegmentAmountInNXMRepriced = getNXMForAssetAmount(
```

Comment:
to be consistent, rename `oldSegmentAmountInNXMRepriced` -> `previousCoverAmountInNXMRepriced`